src/application.coffee

```diff
@@ -644,7 +653,28 @@ application.update = update = (force) ->
 			# Set the updating as finished in its own block, so it never has to worry about other code stopping this.
 			updateStatus.state = UPDATE_IDLE
 
+listenToEvents = ->
+	docker.getEventsAsync()
+	.then (stream) ->
+		parser = JSONStream.parse()
+		stream.pipe(parser)
+		parser.on 'data', (data) ->
```

Comment:
@petrosagg will probably talk about some horrors of using `.on 'data'`, I can't remember the full details myself though